ext/liquid_c/block.c

```diff
@@ -78,6 +101,18 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             {
                 const char *start = token.str + 2, *end = token.str + token.length - 2;
 
+                if (token.str[2] == '-') {
+                    if (RARRAY_LEN(nodelist) > 0) {
+                        VALUE last_node = RARRAY_AREF(nodelist, RARRAY_LEN(nodelist)-1);
+                        rb_funcall(last_node, intern_rstrip, 0);
+                    }
+                    start++;
+                }
+
+                if (token.str[token.length - 3] == '-') {
```

Comment:
Same concern as above, where `{%-%}` can cause the end pointer to come before the start pointer.